src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1067,7 +1077,11 @@ private IMemberInitializerOperation CreateBoundMemberInitializerOperation(BoundA
         {
             Debug.Assert(IsMemberInitializer(boundAssignmentOperator));
 
-            Lazy<IOperation> target = new Lazy<IOperation>(() => Create(boundAssignmentOperator.Left));
+            Lazy<IOperation> target = new Lazy<IOperation>(() =>
+                boundAssignmentOperator.Left.Kind == BoundKind.ObjectInitializerMember && boundAssignmentOperator.Right.Kind == BoundKind.CollectionInitializerExpression
+                    ? CreateBoundObjectInitializerMemberOperation((BoundObjectInitializerMember)boundAssignmentOperator.Left, inCollectionInitializer: true)
```

Comment:
>inCollectionInitializer [](start = 126, length = 23)

The name feels somewhat misleading, the target isn't really inside a collection initializer, it is being initialized with a collection initializer.